membrane_curvature/lib/mods.py

```diff
@@ -160,27 +54,24 @@ def core_fast_leaflet(z_Ref, leaflet, traj, jump, n_cells, lipid_types,
 
     grid_count = np.zeros([n_cells, n_cells])
 
-    for frame in range(0, traj.n_frames, jump):
+    for ts in universe.trajectory:
+
         grid_1 = np.zeros([n_cells, n_cells])
         grid_2 = np.zeros([n_cells, n_cells])
 
         factor = np.float32(n_cells / max_width)
 
-        for lipid_type in lipid_types:
-
-            for bead in lipid_ref[leaflet][lipid_type]:
-
-                x, y, z = traj.xyz[frame, bead, :]
+        for bead in selection:
```

Comment:
Once you start optimizing for performance, you most likely will want to completely eliminate this for-loop.